lib/FrontendTool/FrontendTool.cpp

```diff
@@ -1623,20 +1623,37 @@ int swift::performFrontend(ArrayRef<const char *> Args,
   // arguments are generated by the driver, not directly by a user. The driver
   // is responsible for emitting diagnostics for its own errors. See SR-2683
   // for details.
-  //
-  // FIXME: Do we need one SerializedConsumer per primary? Owned by the
-  // SourceFile?
-  std::unique_ptr<DiagnosticConsumer> SerializedConsumer;
+  std::unique_ptr<DiagnosticConsumer> SerializedConsumerMultiplexer;
```

Comment:
`SerializedConsumerMultiplexer` gives me pause. Thinking aloud... It directs SerializedDiagnostics to the appropriate consumer. In a way, it's a *demultiplexer,* taking a single stream of multiplexed diagnostics and separating them out into separate streams, each directed to a specific consumer. Feels like there could be a better name here.

Reply:
Oops, right. I always get the mux/demux order backwards.

I don't actually care what this name is, and neither does the code; this is only here to keep the consumer alive. I had `TopSerializedConsumer` before but that was really even more vacuous. Any suggestions?

Reply:
Pavel came up with "dispatcher", which I thought was fine. "Distributor" isn't bad either.